src/vm/hints/secp/ec_utils.rs

```diff
@@ -91,6 +91,161 @@ pub fn compute_doubling_slope(
+    );
+
+    //ids.point1
+    let point1_reloc = get_relocatable_from_var_name("point1", ids, vm, hint_ap_tracking)?;
+
+    let (point1_x_d0, point1_x_d1, point1_x_d2, point1_y_d0, point1_y_d1, point1_y_d2) = (
+        get_integer_from_relocatable_plus_offset(&point1_reloc, 0, vm)?,
+        get_integer_from_relocatable_plus_offset(&point1_reloc, 1, vm)?,
+        get_integer_from_relocatable_plus_offset(&point1_reloc, 2, vm)?,
+        get_integer_from_relocatable_plus_offset(&point1_reloc, 3, vm)?,
+        get_integer_from_relocatable_plus_offset(&point1_reloc, 4, vm)?,
+        get_integer_from_relocatable_plus_offset(&point1_reloc, 5, vm)?,
+    );
+
+    let value = line_slope(
+        (
+            pack(point0_x_d0, point0_x_d1, point0_x_d2, &vm.prime),
+            pack(point0_y_d0, point0_y_d1, point0_y_d2, &vm.prime),
+        ),
+        (
+            pack(point1_x_d0, point1_x_d1, point1_x_d2, &vm.prime),
+            pack(point1_y_d0, point1_y_d1, point1_y_d2, &vm.prime),
+        ),
+        &SECP_P,
+    );
+
+    vm.exec_scopes
+        .assign_or_update_variable("value", PyValueType::BigInt(value.clone()));
+
+    vm.exec_scopes
+        .assign_or_update_variable("slope", PyValueType::BigInt(value));
+
+    Ok(())
+}
+
+/*
+Implements hint:
+%{
+    from starkware.cairo.common.cairo_secp.secp_utils import SECP_P, pack
+
+    slope = pack(ids.slope, PRIME)
+    x = pack(ids.point.x, PRIME)
+    y = pack(ids.point.y, PRIME)
+
+    value = new_x = (pow(slope, 2, SECP_P) - 2 * x) % SECP_P
+%}
+*/
+pub fn ec_double_assign_new_x(
+    vm: &mut VirtualMachine,
+    ids: &HashMap<String, BigInt>,
+    hint_ap_tracking: Option<&ApTracking>,
+) -> Result<(), VirtualMachineError> {
+    //ids.slope
+    let slope_reloc = get_relocatable_from_var_name("slope", ids, vm, hint_ap_tracking)?;
+
+    let (slope_d0, slope_d1, slope_d2) = (
+        get_integer_from_relocatable_plus_offset(&slope_reloc, 0, vm)?,
+        get_integer_from_relocatable_plus_offset(&slope_reloc, 1, vm)?,
+        get_integer_from_relocatable_plus_offset(&slope_reloc, 2, vm)?,
+    );
+
+    //ids.point
+    let point_reloc = get_relocatable_from_var_name("point", ids, vm, hint_ap_tracking)?;
+
+    let (x_d0, x_d1, x_d2, y_d0, y_d1, y_d2) = (
+        get_integer_from_relocatable_plus_offset(&point_reloc, 0, vm)?,
+        get_integer_from_relocatable_plus_offset(&point_reloc, 1, vm)?,
+        get_integer_from_relocatable_plus_offset(&point_reloc, 2, vm)?,
+        get_integer_from_relocatable_plus_offset(&point_reloc, 3, vm)?,
+        get_integer_from_relocatable_plus_offset(&point_reloc, 4, vm)?,
+        get_integer_from_relocatable_plus_offset(&point_reloc, 5, vm)?,
+    );
+
+    let slope = pack(slope_d0, slope_d1, slope_d2, &vm.prime);
+    let x = pack(x_d0, x_d1, x_d2, &vm.prime);
+    let y = pack(y_d0, y_d1, y_d2, &vm.prime);
+
+    let value = (slope.pow(2).mod_floor(&SECP_P) - (&x * 2_usize)).mod_floor(&SECP_P);
```

Comment:
Modulus is transitive to substraction, shift by 1 is cheaper than multiplication by 2:
```suggestion
    let value = (slope.pow(2) - (&x << 1_usize)).mod_floor(&SECP_P);
```

Reply:
I knew you would find a better way to do this, but I could figure out a better one :P 
Done